aws/rds-customer-cluster/main.tf

```diff
@@ -0,0 +1,545 @@
+  preferred_backup_window          = var.preferred_backup_window
+  preferred_maintenance_window     = var.preferred_maintenance_window
+  port                             = var.port
+  db_subnet_group_name             = "mattermost-provisioner-db-${var.secondary_vpc_id}-postgresql"
+  vpc_security_group_ids           = [data.aws_security_group.db_sg_secondary[0].id]
+  storage_encrypted                = var.storage_encrypted
+  apply_immediately                = var.apply_immediately
+  db_cluster_parameter_group_name  = aws_rds_cluster_parameter_group.cluster_parameter_group_postgresql_secondary[0].id
+  db_instance_parameter_group_name = aws_db_parameter_group.db_parameter_group_postgresql_secondary[0].id
+  copy_tags_to_snapshot            = var.copy_tags_to_snapshot
+  snapshot_identifier              = var.creation_snapshot_arn_secondary == "" ? null : var.creation_snapshot_arn_secondary
+  allow_major_version_upgrade      = var.allow_major_version_upgrade
+  enabled_cloudwatch_logs_exports  = var.enabled_cloudwatch_logs_exports
+
+
+  tags = merge(
+    {
+      "Counter"                             = 0,
+      "MultitenantDatabaseID"               = format("rds-cluster-multitenant-%s-%s-secondary", split("-", var.secondary_vpc_id)[1], local.database_id),
+      "VpcID"                               = var.secondary_vpc_id,
+      "DatabaseType"                        = var.multitenant_tag,
+      "MattermostCloudInstallationDatabase" = "PostgreSQL/Aurora"
+    },
+    var.tags
+  )
+  lifecycle {
+    ignore_changes = [
+      tags["Counter"],
+      replication_source_identifier
+    ]
+  }
+}
+
+resource "aws_rds_cluster_instance" "provisioning_rds_db_instance_secondary" {
+  count                           = var.enable_global_cluster ? var.replica_min : 0
+  provider                        = aws.secondary
+  identifier                      = format("rds-db-instance-multitenant-%s-%s-sec-%s", split("-", var.secondary_vpc_id)[1], local.database_id, (count.index + 1))
+  cluster_identifier              = aws_rds_cluster.provisioning_rds_cluster_secondary[0].id
+  engine                          = var.engine
+  engine_version                  = var.engine_version
+  instance_class                  = var.instance_type
+  db_subnet_group_name            = "mattermost-provisioner-db-${var.secondary_vpc_id}-postgresql"
+  db_parameter_group_name         = aws_db_parameter_group.db_parameter_group_postgresql_secondary[0].id
+  preferred_maintenance_window    = var.preferred_maintenance_window
+  apply_immediately               = var.apply_immediately
+  monitoring_role_arn             = data.aws_iam_role.enhanced_monitoring.arn
+  monitoring_interval             = var.monitoring_interval
+  promotion_tier                  = count.index + 1
+  performance_insights_enabled    = local.performance_insights_enabled
+  performance_insights_kms_key_id = local.performance_insights_enabled ? var.kms_key_id : ""
```

Comment:
we need to support 2 kms_ids. 1 per region ideally. So I believe we need 2 variables instead of 1. If anyone wants to use 1 they can set the same value in both.

Reply:
Nice one thanks. Fixed ;)